enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,525 @@
+this by annotating their resources, so it's not completely arbitrary.
+
+This will need to be evaluated on a case by case basis as a component considers adding the annotation
+to its resources that will allow it to be filtered out/disabled.
+
+5. What if a user specifies a capability name to exclude that doesn't match anything?
+
+The installer won't know it's not going to match anything, so it should do nothing.  The CVO
+can tell it doesn't match anything, so it can provide a warning on the clusterversion resource, but
+it should not be an error since a user might put a keyword into the list in anticipation of upgrading
+to a new OCP version where the keyword will match (and thus exclude) a new capability, or perhaps
+the keyword used to match a capability that no longer exists after an upgrade.
+
+Along the same lines, if a user both "Excludes" and "Includes" a capability(if we ever add Include as
+a thing you can do), that should be treated as an error/invalid configuration.
+
+6. What to do(if anything) for components with interdependencies, to ensure a user doesn't break
+enabled components by disabling a dependency?  Options include:
+
+* Do nothing other than document dependencies so users know what not to turn off
+* Don't even annotate dependency resources for filtering, so if something is a dependency it cannot be turned off
+* Logic in the install or CVO that intelligently analyzes the filters the user has supplied and checks
+  for dependency issues (least desirable solution imho).
+
+Our current plan is to only annotate resources that are not depended upon (aka "leaf" components)
+
+### Open Questions
+
+1. What to do for components where disabling them has implications on other components or the way certain
+apis behave.  Example: disabling the internal registry changes the behavior of imagestreams
+(can't push to the imagestream anymore to push content to the internal registry) as well as the assumptions
+made by tools like new-app/new-build (create imagestreams that push to the internal registry).
+
+### Test Plan
+
+1) Install clusters w/ the various add-on components included/excluded and confirm
+that the cluster is functional but only running the expected add-ons.  One of these
+tests should disable all optional components to ensure the cluster is functional.
+
+2) Upgrade a cluster to a new version that includes new resources that belong to
+an addon that was included in the original install.  The new resources should be
+created.
+
+3) Upgrade a cluster to a new version that includes new resources that belong to
+an addon that was excluded in the original install.  The new resources should *not* be
+created.
+
+4) After installing a cluster, enable additional addons.  The newly enabled addons should
+be installed/reconciled by the CVO.
+
+5) After installing a cluster, disable an addon.  The configuration change should be
+rejected by the CVO.  Disabling a component post-install is not supported.
+
```

Comment:
@bparees per the 4.11 kick off, if samples operator ends up being the first candidate for this endeavor, to you want to capture the fact that there will be e2e adjustments needed in openshift/origin here in this EP?

or do you feel that is "too detailed" for this EP (or this current iteration of this EP), or do you anticipate that being captured either in another EP, or just captured in Jira items that ultimately accompany the implementation of this EP?

thanks

Reply:
maybe your item 1) is a very high level summary for capturing my point ^^ .... but I know the openshift/origin e2e's is a detail you and I have discussed in the past, so was half expecting some reference to it :-)

Reply:
yeah item (1) was intended to cover that point, but i'll add a sentence:  
```
This will necessitate disabling or modifying tests that depend on the disabled
component(s).
```